src/app/metadata/features/dataset-details-table/dataset-details-table.html

```diff
@@ -0,0 +1,65 @@
+<mat-expansion-panel>
+  <mat-expansion-panel-header>{{ header() }}</mat-expansion-panel-header
+  ><ng-template matExpansionPanelContent
+    ><div class="overflow-auto">
+      <table
+        mat-table
+        class="bold-table-headers w-full min-w-[1280px]"
+        [dataSource]="dataSource()"
+        matSort
+      >
+        <caption class="cdk-visually-hidden">
+          {{ caption() }}
+        </caption>
+        @for (col of columns(); track col){
+        <ng-container [matColumnDef]="col.columnDef">
+          <th mat-header-cell *matHeaderCellDef mat-sort-header [class]="col.class">
+            {{col.header}}
+          </th>
+          <td mat-cell *matCellDef="let item">
+            @let colDef = col.columnDef; @let itemValue = col.childVariable ?
+            item[col.variableName] ? item[col.variableName][col.childVariable] :
+            undefined : item[col.variableName]; @if (colDef === 'phenotype' &&
+            itemValue.constructor.name === "Array") { @for (x of itemValue; track x; let
+            last = $last) { {{x}}{{last ? '' : ', '}} } } @else if (colDef === 'size') {
+            {{itemValue ? (itemValue | parseBytes) : 'N/A'}} } @else if (colDef ===
+            'origin') { {{itemValue | underscoreToSpace}} } @else if (colDef ===
+            'location') { {{itemValue ? storageLabels()[itemValue] || itemValue :
+            'N/A'}} } @else if (colDef === 'hash') { @if (itemValue) {<span
+              class="inline-block max-w-[10ex] overflow-hidden font-mono text-ellipsis whitespace-nowrap"
+              title="{{itemValue}}"
+              >{{itemValue}}</span
+            >
+            <button
```

Comment:
We could add a "copyButton" property to the DatasetDetailsTableColumn and show the button whenever this property is set to true, to make it more universal.